### PR TITLE
fix: remove outdated transactions

### LINKED ISSUE #4044: Approval is stuck pending
**Description**
Initiating approval didn't result with a wallet popup for signing.
Now the interface is stuck indicating pending status and won't allow to initiate a new approval

Clearing local storage didn't help 

image
image

**How to Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**
Also, if it's relevant for the described byg, add information like the browser or OS.

### LINKED ISSUE #4057: Return 'speed-up' and 'cancel' onchain TX feature
Reported and discussed in https://cowservices.slack.com/archives/C0361CDG8GP/p1710427879899379

we lost 'cancel' and 'speed-up' transactions feature that we had 1 year or so. So when a user speeds-up or cancels a TX in the wallet, it will hang in Pending forever state on the CoW Swap UI. Same will be with Approval flow and other onchain transactions. TX hash in the pending log will navigate to 404 page.

image

----

This is how it was implemented (example) https://github.com/cowprotocol/cowswap/pull/1698#issuecomment-1348073416
When a user activated cancel/speed up TX event, we started to show both TXs in the log, and the winning one (cancelled or the original one) remained in the log after execution

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #4057

Pending transaction might get struck because we don't check transactions nonce at all.
Yes, we have `CancelReplaceTxUpdater` which is based on Blocknative API and available only on Mainnet.
This updater is supposed to solve this issue and it solves it, but Blocknative APi is not super reliable and is not available in some countries (e.g. Belarus).

The idea of this fix:
1. Add transactions to the redux store only with `nonce`
2. In `FinalizeTxUpdater` remove a transaction if it is outdated (account nonce is bigger that transaction nonce)

Each Ethereum account has a counter (`nonce`) that means how many transactions were created by this account.

Let's consider an example:
You send a transaction, your account's `nonce` is 3. The transaction hash is `0x0abd...` and it's in pending state.
You decide to cancel/speed up the transaction. To do that you actually just create another transaction with higher fee, so miners will give a precedence to it. That transaction will have a different hash, for example `0x66cad...`.
Once `0x66cad...` transaction is mined, the account `nonce` will be increased and become `4`.

My fix is based on this mechanism.
In CoW Swap we have `FinalizeTxUpdater` which is responsible for updating pending transactions.
On each new block it tries to update pending transactions.
If the updater couldn't successfully update a transaction, I compare current account `nonce` with a transaction `nonce` and if account `nonce` is bigger I just delete the transaction.
The fact that the account nonce is bigger means that the transaction will never be mined because some another transaction was mined first.

The fix has one problem. In `CancelReplaceTxUpdater` we know what happened with a transaction and we display the reason in the transactions history.
Unfortunately, I can't derive a reason in my fix so I just delete the transaction for the history.

image

  # To Test

1. Use Gnosis chain or Sepolia
2. Send an approve transaction (set low gas in your wallet to have a time to cancel the transaction)
3. Cancel/speed up the transaction
- [ ] AR: the transaction get stuck in pending state
- [ ] ER: the transaction disappears once the replacement tx is mined

### CODE REVIEW

**Comment by alfetopito** [comment]
The fix works, however, not very keen on the fact that the pending tx disappears.
Could we in this case, load the tx info for the replacement tx (be it speed up or cancellation) and create a tx entry for it?

**Comment by shoom3301** [comment]
@alfetopito I agree. The only solution I have - "Replaced" status. Unfortunately, I can't distinguish was a transaction cancelled or speed up, so I added a new type "Replaced":

image

**Comment by shoom3301** [comment]
image

Added a title for the status label

**Comment by alfetopito** [comment]
Nice, I like this behaviour much more.

The activity modal has the right state.

The tokens page state is not updated, though

image

Didn't check but probably approval modals inside the forms need to be updated too?

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx [hunk 1]
```diff
@@ -97,8 +98,11 @@ export function StatusDetails(props: StatusDetailsProps) {
         isCancelling={isCancelling}
         isPresignaturePending={isPresignaturePending}
         isCreating={isCreating}
+        title={isReplaced ? 'Transaction was cancelled or speed up' : ''}
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx:101]: This is in the past, so it should be either `sped up` or `speeded up` https://grammarwiki.com/past-tense-of-speed/

I prefer `sped up`.

**Review by alfetopito — APPROVED** [review]
Only one comment about the verb tense

File: apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx]: Thank you! Fixed

**Review by elena-zh — CHANGES_REQUESTED** [review]
Hey @shoom3301 , the #4057 issue is mostly fixed in the current PR!
But I have to request changes on this PR as the fix does not solve the #4044 issue: the Approve button (or any other transactions) hangs in the pending state and is not removed when something goes wrong with the TX updater/wallet connection/etc. 

Additionally, here are the issues would be great to address:
1. When I cancelled the TX or relaced it in the Safe (reported issues #245 and #2393): keep the TX in the eternal pending status
image
2. ETH flow orders (classic and a new one): does not work with cancellation/speeding up (#1704 issue)
3. Convert vCOW card also is not updated when modify a TX: #4093

**Comment by shoom3301** [comment]
@elena-zh #4044 is fixed here:
1. https://github.com/cowprotocol/cowswap/pull/4125
2. https://github.com/cowprotocol/cowswap/pull/4130

**Review by elena-zh — APPROVED** [review]
Approving!
However, it would be great to address #1704 issue (as well as #4138 and #4093)

**Pull request merged**